.github/workflows/main.yaml

```diff
@@ -4,10 +4,12 @@ on:
   pull_request:
     branches: ['*']
   push:
-    branches: ['*']
+    branches: ['master']
 
 # Ensures that only one deploy task per branch/environment will run at a time.
-concurrency: ci-${{ github.ref }}
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
```

Comment:
Wouldn't it be even safer to make this more restrictive, like:
```
  group: ${{ github.workflow }}
```

?
This would keep multiple 'triggers' of this workflow from running concurrently.

Reply:
mehh, i've changed the above `on branches` from `*` to `master` therefore we really only have one branch (or environment) that deploys.